src/app/databaseService.ts

```diff
@@ -139,9 +139,9 @@ async function getTestResultDetails(
   const testResultQueryResult = queryResult[0][0] as TestResultQueryResult;
 
   if (
-    testResultQueryResult === undefined ||
-    testResultQueryResult.id === undefined ||
-    testResultQueryResult.result === undefined
+    testResultQueryResult === undefined
```

Comment:
Maybe add a function to include these checks like  `isResultDefined` or something similar

Reply:
Simplification and unit tests added.